plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -82,6 +83,37 @@ public function __construct() {
 		}
 	}
 
+
+	/**
+	 * Adds the product template ID to the product if it doesn't exist.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WC_Product       $product The product.
+	 */
+	public function possibly_add_template_id( $response, $product ) {
+		if ( ! $product ) {
+			return $response;
+		}
+		if ( ! $product->meta_exists( '_product_template_id' ) ) {
+			/**
+			 * Experimental: Allows to determine a product template id based on the product data.
+			 *
+			 * @ignore
+			 * @since 9.1.0
+			 */
+			$product_template_id = apply_filters( 'experimental_woocommerce_product_editor_determine_product_template', '', $product );
```

Comment:
I think a better name for this filter would be `experimental_woocommerce_product_editor_product_template_for_product`

I don't think verbs like `determine` are needed in a filter name.